substrate/client/transaction-pool/src/graph/pool.rs

```diff
@@ -320,18 +322,22 @@ impl<B: ChainApi, L: EventHandler<B>> Pool<B, L> {
 		// Option<Vec<Tag>>)`)
 		let all = extrinsics.iter().zip(in_pool_tags.into_iter());
 		let mut validated_counter: usize = 0;
-
 		let mut future_tags = Vec::new();
 		let now = Instant::now();
 		for (extrinsic, in_pool_tags) in all {
 			match in_pool_tags {
 				// reuse the tags for extrinsics that were found in the pool
 				Some(tags) => future_tags.extend(tags),
-				// if it's not found in the pool query the runtime at parent block
-				// to get validity info and tags that the extrinsic provides.
 				None => {
-					// Avoid validating block txs if the pool is empty
-					if !self.validated_pool.status().is_empty() {
+					// if it's not found in the pool, check if the extrinsic `provides`
+					// tags are known (from inactive views, queried at an upper level)
+					let xt_hash = self.hash_of(extrinsic);
+					if let Some(tags) = known_provides_tags.get(&xts_hash) {
+						future_tags.extend(tags.clone());
+						log::trace!(target: LOG_TARGET,"[{:?}] prune::inactive_view_tags {:?}", xts_hash, tags);
+					} else if !self.validated_pool.status().is_empty() {
+						// and if that's not the case, query the runtime at parent block
+						// to get validity info and tags that the extrinsic provides.
```

Comment:
known_provided_tags could be merged with in_pool_tags to avoid if here. it could also be filtered to contain only needed hashes.

wdyt?